uvtools/dspec.py

```diff
@@ -60,11 +60,356 @@ def calc_width(filter_size, real_delta, nsamples):
         lthresh = nsamples
     return (uthresh, lthresh)
 
-
+def fourier_filter(x, data, wgts, filter_centers, filter_half_widths, suppression_factors,
+                   mode, filter2d, fitting_options, cache=None, filter_dim=1, skip_wgt=0.1,
+                   max_contiguous_edge_flags=10):
+                   '''
+                   Your one-stop-shop for fourier filtering.
+                   We don't use the other filtering functions anymore.
```

Comment:
this comment is not relevant if the other funcs are deprecated.

Reply:
I think we can hold off on deprecation. I went in hoping to deprecate since (mostly my fault) high_pass_fourier_filter was turning into a mess since I tried to expand it far beyond its original intent (it was only originally supposed to support CLEAN but when I added more filtering modes, arguments exploded). I think we can decide whether we want to deprecate down the road if we find that there isn't any functionality left.